docs/_posts/2015-12-09-ismounted-antipattern.md

```diff
@@ -0,0 +1,55 @@
+---
+title: "isMounted is an Antipattern"
+author: jimfb
+---
+
+The primary use case for `isMounted()` is to avoid calling `setState()` after a component has unmounted, because calling `setState()` after a component has unmounted will emit a warning. The “setState warning” exists to help you catch bugs, because calling `setState()` on an unmounted component is an indication that your app/component has somehow failed to clean up properly. Specifically, calling `setState()` in an unmounted component means that your app is still holding a reference to the component after the component has been unmounted - which often indicates a memory leak!
+
+To avoid the error message, people often add lines like this:
+
+```js
+if(this.isMounted()) this.setState({...}); // This is bad.
+```
+
+Checking `isMounted` before calling `setState()` does eliminate the warning, but it also defeats the purpose of the warning, since now you will never get the warning (even when you should!)
+
+Most other uses of `isMounted()` are similarly erroneous; using `isMounted()` is usually a code smell.
+
+An easy migration strategy for anyone upgrading their code to avoid `isMounted()` is to track the mounted status yourself.  Just set a `_isMounted` property to true in `componentDidMount` and set it to false in `componentWillUnmount`, and use this variable to check your component's status.
+
+An optimal solution would be to find places where `setState()` might be called after a component has unmounted, and fix them. Such situations most commonly occur due to callbacks, when a component is waiting for some data and gets unmounted before the data arrives. Ideally, any callbacks should be canceled in `componentWillUnmount`, prior to unmounting.
+
+If you use ES6 promises, you may need to wrap your promise in order to make it cancelable.
```

Comment:
What's the point of this post? Are you discussing why isMounted is an antipattern and some alternative ways to achieve the same result? Are you deprecating isMounted and discussing alternatives? Are you sharing a way to make promises cancelable. I see all 3 of these coming through, but no clear purpose of the post.

If your target audience is everybody, then I think you need a little bit more to give everybody something to relate to. We don't want people to dismiss the content because half the page is taken up by a block of code for Promises.

You can say "trivial" all you want but it's important to remember that this stuff isn't immediately obvious to everybody.